lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_send_app_size_metrics.rb

```diff
@@ -0,0 +1,165 @@
+            type: FastlaneCore::Boolean,
+            default_value: true
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :app_name,
+            env_name: 'FL_IOS_SEND_APP_SIZE_METRICS_APP_NAME',
+            description: 'The name of the app for which we are publishing metrics, to help filter by app in the dashboard',
+            type: String,
+            optional: false
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :app_version,
+            env_name: 'FL_IOS_SEND_APP_SIZE_METRICS_APP_VERSION',
+            description: 'The version of the app for which we are publishing metrics, to help filter by version in the dashboard',
+            type: String,
+            optional: false
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :build_type,
+            env_name: 'FL_IOS_SEND_APP_SIZE_METRICS_BUILD_TYPE',
+            description: 'The build configuration for which we are publishing metrics, to help filter by build config in the dashboard. E.g. `Debug`, `Release`',
+            type: String,
+            optional: true
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :source,
+            env_name: 'FL_IOS_SEND_APP_SIZE_METRICS_SOURCE',
+            description: 'The type of event at the origin of that build, to help filter data in the dashboard. E.g. `pr`, `beta`, `final-release`',
+            type: String,
+            optional: true
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :ipa_path,
+            env_name: 'FL_IOS_SEND_APP_SIZE_METRICS_IPA_PATH',
+            description: 'The path to the .ipa to extract size information from',
+            type: String,
+            optional: false,
+            default_value: Actions.lane_context[SharedValues::IPA_OUTPUT_PATH],
+            verify_block: proc do |value|
+              UI.user_error!('You must provide an path to an existing `.ipa` file') unless File.exist?(value)
+            end
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :app_thinning_plist_path,
+            env_name: 'FL_IOS_SEND_APP_SIZE_METRICS_APP_THINNING_PLIST_PATH',
+            description: 'The path to the `app-thinning.plist` file to extract thinning size information from. ' \
+              + 'By default, will try to use the `app-thinning.plist` file next to the `ipa_path`, if that file exists',
+            type: String,
+            optional: true,
+            default_value_dynamic: true
+          ),
+        ]
+      end
+
+      def self.return_type
+        :integer
+      end
+
+      def self.return_value
+        'The HTTP return code from the call. Expect a 201 when new metrics were received successfully and entries created in the database'
+      end
+
+      def self.authors
+        ['automattic']
+      end
+
+      def self.is_supported?(platform)
+        [:ios, :mac].include? platform
```

Comment:
Is macOS appropriate here, when the action expect `.ipa` files?

```suggestion
        platform == :ios
```

Reply:
Good point. I have no idea if there's a similar concept of app thinning with macOS `.app` files (would probably make sense, as in Universal Binaries vs Intel-only vs M1-only…?)… but since we haven't tested it and the code wasn't written with this use case in mind so probably might need some tweaks (and specific unit tests!) to make sure we'd support macOS… probably better to not declare the support for now anyway 👍